genienlp/validate.py

```diff
@@ -81,21 +96,27 @@ def generate_with_model(model, data_iterator, numericalizer, task, args, predict
             batch_context = numericalizer.reverse(batch.context.value.data, task=task, field_name='context')
             contexts += batch_context
         predictions += batch_prediction
+        confidences += batch_confidences
     
     if original_order is not None:
         # sort back to the original order
-        original_order, example_ids, predictions, answers, contexts = tuple(zip(*sorted(list(zip(original_order, example_ids, predictions, answers, contexts)))))
-
-    if prediction_file_name is not None:
-        with open(prediction_file_name, 'w' + ('' if args.overwrite else 'x')) as prediction_file:
-            for i in range(len(example_ids)):
-                prediction_file.write(example_ids[i] + '\t' + '\t'.join(predictions[i]) + '\n') # write all outputs in the prediction file, separated by \t
+        original_order, example_ids, predictions, answers, contexts, confidences = [list(a) for a in tuple(zip(*sorted(list(zip(original_order, example_ids, predictions, answers, contexts, confidences)))))]
     
-    if output_predictions_only:
-        return predictions
     # TODO calculate and return loss
     loss = None
-    return loss, predictions, answers, contexts
+    output = ()
+
+    if output_predictions_only:
+        output = (predictions, )
+    else:
+        output = (loss, example_ids, predictions, answers, contexts)
+    if output_confidences:
+        output = output + (confidences, )
+    if output_confidence_scores:
+        confidence_scores = confidence_estimator.estimate(confidences)
+        output = output + (confidence_scores, )
+
+    return output
```

Comment:
I wonder if it's time to switch to a dataclass or namedtuple for this, it's a pretty big tuple already...